modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/passthru/PassThroughHttpSenderTest.java

```diff
@@ -0,0 +1,133 @@
+
+    @Test
+    public void testInit() throws Exception {
+        ConfigurationContext configurationContext = new ConfigurationContext(new AxisConfiguration());
+        WorkerPool workerPool = new NativeWorkerPool(3, 4, 5, 5, "name", "id");
+        configurationContext.setProperty(PassThroughConstants.PASS_THROUGH_TRANSPORT_WORKER_POOL, workerPool);
+        TransportOutDescription transportOutDescription = new TransportOutDescription("passthru");
+
+        PassThroughHttpSender passThroughHttpSender = new PassThroughHttpSender();
+
+        passThroughHttpSender.init(configurationContext, transportOutDescription);
+    }
+
+    @Test
+    public void testInvoke() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        MessageContext messageContext = new MessageContext();
+        messageContext.setProperty(Constants.Configuration.TRANSPORT_URL, "http://sample.url");
+        messageContext.setProperty(NhttpConstants.FORCE_HTTP_CONTENT_LENGTH, true);
+        messageContext.setProperty(PassThroughConstants.COPY_CONTENT_LENGTH_FROM_INCOMING, true);
+        messageContext.setProperty(PassThroughConstants.ORGINAL_CONTEN_LENGTH, "1000");
+        messageContext.setOperationContext(new OperationContext());
+
+        sender = PowerMockito.spy(sender);
+
+        PowerMockito.when(targetConfiguration.getBufferFactory()).thenReturn(factory);
+        PowerMockito.doNothing().when(sender, "sendRequestContent", any(MessageContext.class));
+
+        Handler.InvocationResponse response = sender.invoke(messageContext);
+
+        Assert.assertNotNull("PassThrough Http Sender not invoked!", response);
+    }
+
+    @Test
+    public void testSubmitResponse() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        MessageContext messageContext = new MessageContext();
+        messageContext.setProperty(Constants.Configuration.TRANSPORT_URL, "http://sample.url");
+        messageContext.setProperty(NhttpConstants.FORCE_HTTP_CONTENT_LENGTH, true);
+        messageContext.setProperty(PassThroughConstants.COPY_CONTENT_LENGTH_FROM_INCOMING, true);
+        messageContext.setProperty(PassThroughConstants.ORGINAL_CONTEN_LENGTH, "1000");
+        messageContext.setOperationContext(new OperationContext());
+
+        IOSession session = PowerMockito.mock(IOSession.class);
+        NHttpServerConnection conn = new DefaultNHttpServerConnection(session, 12);
+
+        messageContext.setProperty(PassThroughConstants.PASS_THROUGH_SOURCE_CONNECTION, conn);
+        messageContext.setProperty(PassThroughConstants.HTTP_ETAG_ENABLED, true);
+
+        Map headers = new HashMap();
+        headers.put("randomKey", "randomValue");
+
+        messageContext.setProperty(MessageContext.TRANSPORT_HEADERS, headers);
+
+        sender = PowerMockito.spy(sender);
+
+        PowerMockito.when(digestGenerator.getDigest(any(MessageContext.class))).thenReturn("testString");
+        sender.submitResponse(messageContext);
```

Comment:
any reason why we don't have a assert in this test? (if its a testcase what are we testing)

Reply:
The submit-response method does not return anything but print some logs. The method completes only if there are no errors. Therefore, it's assumed that if the method completes, the test case passes.  Otherwise we'll have to have a log-viewer client as in the integration tests and check for logs.